File: Backend/app/models/onboarding.py
from pydantic import BaseModel


class OnboardingUserData(BaseModel):
    first_name: str | None = None
    last_name: str | None = None
    phone: str | None = None


class OnboardingBusinessData(BaseModel):
    name: str
    business_format: str | None = None
    description: str | None = None
    brand_color: str | None = None
    b2b_or_b2c: str | None = None
    website_url: str | None = None
    instagram_handle: str | None = None
    target_location: str | None = None
    ideal_customer: str | None = None
    products_services: str | None = None
    industry_niche: str | None = None


class OnboardingRequest(BaseModel):
    user: OnboardingUserData | None = None
    business: OnboardingBusinessData
    plan_id: str | None = None


class OnboardingResponse(BaseModel):
    user_id: str
    business_id: str
    plan_id: str | None = None
    message: str = "Onboarding completed successfully"

File: Frontend/src/api/onboarding.ts
import { api } from './client';

interface OnboardingUserPayload {
  first_name?: string;
  last_name?: string;
  phone?: string;
}

interface OnboardingBusinessPayload {
  name: string;
  business_format?: string;
  description?: string;
  brand_color?: string;
  b2b_or_b2c?: string;
  website_url?: string;
  instagram_handle?: string;
  target_location?: string;
  ideal_customer?: string;
  products_services?: string;
  industry_niche?: string;
}

interface OnboardingPayload {
  user?: OnboardingUserPayload;
  business: OnboardingBusinessPayload;
  plan_id?: string;
}

interface OnboardingResponse {
  user_id: string;
  business_id: string;
  plan_id?: string;
  message: string;
}

export const onboardingApi = {
  complete: (data: OnboardingPayload) =>
    api.post<OnboardingResponse>('/onboarding/', data),
};
